Code/Source/RGLSystemComponent.cpp

```diff
@@ -121,14 +135,35 @@ namespace RGL
         return m_sceneConfig;
     }
 
+    static bool HasExcludedTag(AZ::EntityId entityId, const AZStd::vector<LmbrCentral::Tag>& excludedTags)
+    {
+        LmbrCentral::Tags entityTags;
+        LmbrCentral::TagComponentRequestBus::EventResult(entityTags, entityId, &LmbrCentral::TagComponentRequests::GetTags);
+
+        if (entityTags.empty())
+        {
+            return false;
+        }
+
+        return AZStd::any_of(
+            excludedTags.cbegin(),
+            excludedTags.cend(),
+            [&entityTags](const auto& tag)
+            {
+                return entityTags.contains(tag);
+            });
+    }
+
     void RGLSystemComponent::OnEntityContextCreateEntity(AZ::Entity& entity)
     {
-        if (m_excludedEntities.contains(entity.GetId()))
+        if (!HasVisuals(entity))
         {
             return;
         }
 
-        if (m_activeLidarCount < 1U)
+        m_entityTagListeners.emplace_back(entity.GetId());
```

Comment:
`EntityTagListeners` are never erased. Their destruction should be in `OnEntityContextDestroyEntity()` method.